atak/ATAK/app/src/main/java/com/atakmap/android/cot/bluetooth/BluetoothSerialCoTReader.java

```diff
@@ -0,0 +1,62 @@
+package com.atakmap.android.cot.bluetooth;
+
+import android.bluetooth.BluetoothDevice;
+
+import com.atakmap.android.bluetooth.BluetoothASCIIClientConnection;
+import com.atakmap.android.bluetooth.BluetoothConnection;
+import com.atakmap.android.bluetooth.BluetoothCotManager;
+import com.atakmap.android.bluetooth.BluetoothReader;
+import com.atakmap.android.maps.MapView;
+import com.atakmap.coremap.log.Log;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Manages the connection and reading from specific supported CoT over LoRaWAN to Bluetooth providers.
+ */
+public class BluetoothSerialCoTReader extends BluetoothReader {
+
+    private static final String TAG = "BluetoothSerialCoTReader";
+    private BluetoothSerialCoTManager bgcm;
+
+    public BluetoothSerialCoTReader(BluetoothDevice device) {
+        super(device);
+    }
+
+    void register(BluetoothSerialCoTManager bgcm) {
+        this.bgcm = bgcm;
+    }
+
+    @Override
+    public BluetoothCotManager getCotManager(MapView mapView) {
+        BluetoothDevice device = connection.getDevice();
+        String deviceName = device.getName();
+
+        // ATAK-8437 BluetoothGPSNMEAReader NPE
+        // I have never observed a case where the deviceName is null.
```

Comment:
Thank you for pointing that out.    We have observed on some phones specifically with earlier versions of Android that the device getName call would indeed return null.